File: src/main/java/com/webappdevelopment/service/ValuationTeamService.java
package com.webappdevelopment.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.webappdevelopment.domain.ValuationTeam;
import com.webappdevelopment.dto.ValuationTeamDTO;
import com.webappdevelopment.dto.ValuationTeamSearchDTO;
import com.webappdevelopment.dto.ValuationTeamPageDTO;
import com.webappdevelopment.dto.ValuationTeamConvertCriteriaDTO;
import com.webappdevelopment.service.GenericService;
import com.webappdevelopment.dto.common.RequestDTO;
import com.webappdevelopment.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ValuationTeamService extends GenericService<ValuationTeam, Integer> {

	List<ValuationTeam> findAll();

	ResultDTO addValuationTeam(ValuationTeamDTO valuationTeamDTO, RequestDTO requestDTO);

	ResultDTO updateValuationTeam(ValuationTeamDTO valuationTeamDTO, RequestDTO requestDTO);

    Page<ValuationTeam> getAllValuationTeams(Pageable pageable);

    Page<ValuationTeam> getAllValuationTeams(Specification<ValuationTeam> spec, Pageable pageable);

	ResponseEntity<ValuationTeamPageDTO> getValuationTeams(ValuationTeamSearchDTO valuationTeamSearchDTO);
	
	List<ValuationTeamDTO> convertValuationTeamsToValuationTeamDTOs(List<ValuationTeam> valuationTeams, ValuationTeamConvertCriteriaDTO convertCriteria);

	ValuationTeamDTO getValuationTeamDTOById(Integer valuationTeamId);







}

File: webappdevelopment_ui/src/services/ValuationTeamService.js
import http from "../http-common"; 

class ValuationTeamService {
  getAllValuationTeams(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/valuationTeam/valuationTeams`, searchDTO);
  }

  get(valuationTeamId) {
    return this.getRequest(`/valuationTeam/${valuationTeamId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/valuationTeam?field=${matchData}`, null);
  }

  addValuationTeam(data) {
    return http.post("/valuationTeam/addValuationTeam", data);
  }

  update(data) {
  	return http.post("/valuationTeam/updateValuationTeam", data);
  }
  
  uploadImage(data,valuationTeamId) {
  	return http.postForm("/valuationTeam/uploadImage/"+valuationTeamId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ValuationTeamService();
